packages/client/components/JiraScopingSearchBar.tsx

```diff
@@ -1,55 +1,123 @@
-import styled from '@emotion/styled'
 import graphql from 'babel-plugin-relay/macro'
 import React from 'react'
-import {useFragment} from 'react-relay'
-import {PALETTE} from '~/styles/paletteV3'
+import {commitLocalUpdate, useFragment} from 'react-relay'
+import useAtmosphere from '../hooks/useAtmosphere'
+import PersistJiraSearchQueryMutation from '../mutations/PersistJiraSearchQueryMutation'
+import SearchQueryId from '../shared/gqlIds/SearchQueryId'
 import {JiraScopingSearchBar_meeting$key} from '../__generated__/JiraScopingSearchBar_meeting.graphql'
-import JiraScopingSearchCurrentFilters from './JiraScopingSearchCurrentFilters'
 import JiraScopingSearchFilterToggle from './JiraScopingSearchFilterToggle'
-import JiraScopingSearchHistoryToggle from './JiraScopingSearchHistoryToggle'
-import JiraScopingSearchInput from './JiraScopingSearchInput'
-
-const SearchBar = styled('div')({
-  padding: 16
-})
-
-const SearchBarWrapper = styled('div')({
-  alignItems: 'center',
-  border: `1px solid ${PALETTE.SLATE_400}`,
-  borderRadius: '40px',
-  display: 'flex',
-  height: 44,
-  padding: '0 16px',
-  width: '100%'
-})
+import ScopingSearchBar from './ScopingSearchBar'
+import ScopingSearchHistoryToggle from './ScopingSearchHistoryToggle'
+import ScopingSearchInput from './ScopingSearchInput'
+
 interface Props {
   meetingRef: JiraScopingSearchBar_meeting$key
 }
 
 const JiraScopingSearchBar = (props: Props) => {
   const {meetingRef} = props
+  const atmosphere = useAtmosphere()
 
   const meeting = useFragment(
     graphql`
       fragment JiraScopingSearchBar_meeting on PokerMeeting {
-        ...JiraScopingSearchHistoryToggle_meeting
-        ...JiraScopingSearchInput_meeting
         ...JiraScopingSearchFilterToggle_meeting
-        ...JiraScopingSearchCurrentFilters_meeting
+        id
+        teamId
+        jiraSearchQuery {
+          projectKeyFilters
+          queryString
+          isJQL
+        }
+        viewerMeetingMember {
+          teamMember {
+            integrations {
+              atlassian {
+                jiraSearchQueries {
+                  id
+                  queryString
+                  isJQL
+                  projectKeyFilters
+                }
+                projects {
+                  id
+                  name
+                }
+              }
+            }
+          }
+        }
       }
     `,
     meetingRef
   )
 
+  const {id: meetingId, teamId} = meeting
+  const {jiraSearchQueries} = meeting.viewerMeetingMember?.teamMember.integrations?.atlassian ?? {}
+  const searchQueries =
+    jiraSearchQueries?.map((jiraSearchQuery) => {
+      const {id, queryString, isJQL, projectKeyFilters} = jiraSearchQuery
+
+      const selectQuery = () => {
```

Comment:
-2 I like that `ScopingSearchHistoryToggle` is now a reusable component, but the problem is that this complex `searchQueries` logic is now taking place in the `JiraScopingSearchBar`. This means that:

1. On every keystroke in the search input, we run through this logic which contains a nested `map`, `filter`, `indexOf` and creates a few functions. The number of saved `jiraSearchQueries` will probably be fairly small, but if a user is searching & selecting issues every week for over a year, they could have hundreds of saved queries. 
2. The rest of the search bar doesn't need to know about the `searchQueries`. I'd say that it's easier to read the component & component tree if each component only contains the minimum amount of data it needs.

I'd be in favour of reverting back to `JiraScopingSearchHistoryToggle` and keeping the logic there, but feel free to disagree!

Reply:
@nickoferrall Re-render happens because we use relay store as a state and request a search query in a parent fragment.
```
jiraSearchQuery {
  projectKeyFilters
  queryString
  isJQL
}
```
I wrapped `ScopingSearchHistoryToggle` and `ScopingSearchBar` into their own `Jira` prefixed components with their own fragments. It solved the issue.
In my next PR I'm going to try to add more refactoring to `ScopingSearchHistoryToggle` as we are going to use a single table for all search queries.

I'm curious, do we really need to use this approach with storing a search query in the relay store?

Reply:
@nickoferrall I was thinking about using a react context, feels like it could be less hacky, though I need to check. What do you think?

Reply:
I remember @mattkrick saying that React Context is nice and easy to use, but the way it's implemented isn't great for performance. 

I now quite like being able to query global state in Relay in the same way that we query the db, but I do remember finding it confusing when I first discovered the pattern. 

My main concern with putting the search query in React Context is we'd be storing global state in two places which could be confusing.